src/steps/PaymentMethod/index.tsx

```diff
@@ -20,6 +20,13 @@ const PaymentMethod: React.FC<Props> = () => {
     variables: { chainId },
   });
 
+  useEffect(() => {
+    state.provider?.data?.off?.('network');
+    dispatch({ type: 'SET_ACCOUNT_IDLE' });
+    dispatch({ type: 'SET_PROVIDER_IDLE' });
+    dispatch({ payload: undefined, type: 'SET_PROVIDER_CHAIN_ID' });
+  }, []);
```

Comment:
resets provider and account states so that switch chain logic does not get stale